operator/v1/types_cloudcredential.go

```diff
@@ -0,0 +1,63 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudCredential provides a means to configure an operator to manage CredentialsRequests.
+type CloudCredential struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec CloudCredentialSpec `json:"spec,omitempty"`
+	// +optional
+	Status CloudCredentialStatus `json:"status,omitempty"`
+}
+
+// CloudCredentialsMode is the specified mode the cloud-credential-operator
+// should reconcile CredentialsRequest with
+// +kubebuilder:validation:Enum="";manual;mint;passthrough
+type CloudCredentialsMode string
+
+const (
+	// ModeManual tells cloud-credential-operator to not reconcile any CredentialsRequests
+	// (primarily used for the disconnected VPC use-cases).
+	ModeManual CloudCredentialsMode = "manual"
```

Comment:
Should these mode names be more specific to CloudCredential to avoid conflict with some other enum in the package?
```suggestion
	CloudCredentialModeManual CloudCredentialsMode = "manual"
```

Reply:
Pushed out the suggested changes. FWIW, I named it "CloudCredentialsMode" (with the plural CloudCredental**s**).